### PR TITLE
New Documentation: RESTApi Roles endpoints

### LINKED ISSUE #888: Add missing documentation for "roles.list", "roles.create" and "roles.addUserToRole" RESTApi endpoints
Working on it.

### PR DESCRIPTION
**Pull request by aferreira44**
Close #888

Add new documentation for new Roles RESTApi endpoints (https://github.com/RocketChat/Rocket.Chat/issues/11566)

### CODE REVIEW

File: developer-guides/rest-api/roles/addUserToRole/README.md [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    "role": {
+        "_id": "guest",
+        "name": "guest",
+        "scope": "Subscriptions",
+        "description": "",
+        "_updatedAt": "2018-08-23T03:57:54.603Z"
+    },
+    "success": true
+}
+```
```
Comment (MarcosSpessatto) [inline developer-guides/rest-api/roles/addUserToRole/README.md:42]: @aferreira44  Can you add changelog? Like https://github.com/RocketChat/docs/blob/master/developer-guides/rest-api/rooms/get/README.md#change-log

File: developer-guides/rest-api/roles/create/README.md [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    "role": {
+        "_id": "newRole",
+        "name": "newRole",
+        "scope": "Subscriptions",
+        "description": "Role description",
+        "_updatedAt": "2018-08-23T03:57:54.603Z"
+    },
+    "success": true
+}
+```
```
Comment (MarcosSpessatto) [inline developer-guides/rest-api/roles/create/README.md:42]: @aferreira44 changelog here too.

File: developer-guides/rest-api/roles/list/README.md [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+        },
+        {
+            "_id": "livechat-guest",
+            "name": "livechat-guest",
+            "scope": "Users"
+        }
+    ],
+    "success": true
+}
+```
```
Comment (MarcosSpessatto) [inline developer-guides/rest-api/roles/list/README.md:88]: @aferreira44 changelog here too.

**Review by MarcosSpessatto — CHANGES_REQUESTED** [review]

**Comment by MarcosSpessatto** [comment]
@aferreira44 can you please create a section to `roles` and add these endpoints? Like https://github.com/RocketChat/docs/tree/master/developer-guides/rest-api#permissions

File: developer-guides/rest-api/roles/addUserToRole/README.md (hunk 1 above)
Comment (aferreira44) [reply developer-guides/rest-api/roles/addUserToRole/README.md:42]: Done.

File: developer-guides/rest-api/roles/create/README.md (hunk 2 above)
Comment (aferreira44) [reply developer-guides/rest-api/roles/create/README.md:42]: Done.

File: developer-guides/rest-api/roles/list/README.md (hunk 3 above)
Comment (aferreira44) [reply developer-guides/rest-api/roles/list/README.md:88]: Done.

**Comment by aferreira44** [comment]
@MarcosSpessatto Changes done. Can you review them?

**Review by MarcosSpessatto — APPROVED** [review]

**Pull request merged**